src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -214,129 +429,190 @@ performFetchWithCompletionHandler:(void (^)(UIBackgroundFetchResult result))comp
-}
-@end
-```
+Calling `run` on a `FlutterEngine` by default runs the `main()` Dart function
+of your `lib/main.dart` file.
 
-Or, using Swift:
+You can also run a different entrypoint function by using [`runWithEntrypoint`]({{site.api}}/objcdoc/Classes/FlutterEngine.html#/c:objc(cs)FlutterEngine(im)runWithEntrypoint:)
+with an `NSString` specifying a different Dart function.
 
-`ViewController.swift`:
+{{site.alert.note}}
+Dart entrypoint functions other than `main()` need to be annotated with
 
-```swift
-import UIKit
-import Flutter
+<?code-excerpt "main.dart" title?>
+```dart
+@pragma('vm:entry-point')
+void myOtherEntrypoint() { ... };
+```
 
-class ViewController: UIViewController {
-  override func viewDidLoad() {
-    super.viewDidLoad()
-    let button = UIButton(type:UIButton.ButtonType.custom)
-    button.addTarget(self, action: #selector(handleButtonAction), for: .touchUpInside)
-    button.setTitle("Press me", for: UIControl.State.normal)
-    button.frame = CGRect(x: 80.0, y: 210.0, width: 160.0, height: 40.0)
-    button.backgroundColor = UIColor.blue
-    self.view.addSubview(button)
-  }
+in order to not be tree-shaken away when compiling.
+{{site.alert.end}}
 
-  @objc func handleButtonAction() {
-    if let flutterEngine = (UIApplication.shared.delegate as? AppDelegate)?.flutterEngine {
-      let flutterViewController = FlutterViewController(engine: flutterEngine, nibName: nil, bundle: nil)
-      self.present(flutterViewController, animated: false, completion: nil)
-    }
-  }
-}
-```
+### Dart library
 
-You should now be able to build and launch MyApp on the Simulator or on a device.
-Pressing the button should bring up a full-screen Flutter view with the standard
-Flutter Demo counting app. You can use routes to show different widgets at different
-places in your app, as described in the Android section above. To set the route, call
+In addition to specifying a Dart function, you can also specify an entrypoint
+function in a specific file.
 
-- Objective-C:
+For instance,
 
-```objective-c
-[flutterViewController setInitialRoute:@"route1"];
+<?code-excerpt "Objective-C" title?>
+```objectivec
+[flutterEngine runWithEntrypoint:@"myOtherEntrypoint" libraryURI:@"other_file.dart"];
 ```
 
-- Swift:
+or
 
+<?code-excerpt "Swift" title?>
 ```swift
-flutterViewController.setInitialRoute("route1")
+flutterEngine.run(withEntrypoint: "myOtherEntrypoint", libraryURI: "other_file.dart")
 ```
 
-immediately after construction of the `FlutterViewController` (and before presenting
-it).
+will run `myOtherEntrypoint()` in `lib/other_file.dart` instead of `main()` in
+`lib/main.dart`.
 
-You can have the Flutter app dismiss itself by calling `SystemNavigator.pop()`
-in the Dart code.
+### Route
 
-## Building and running your app
+An initial route can be set for your Flutter [`WidgetsApp`]({{site.api}}/flutter/widgets/WidgetsApp-class.html)
+when constructing the engine.
```

Comment:
`If you choose to do this, you _must_ do so before running the engine, and any calls after running the engine will be ignored.`

IMO, we should avoid documenting this API, deprecate it, and remove it.  It's the source of a lot of confusion and it has no clear fix.

Reply:
Added a warning box.

Reply:
Actually there is already a warning box.